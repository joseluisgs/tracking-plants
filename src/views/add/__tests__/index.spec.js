// Para hacer una copia del componente montado
import { shallowMount } from '@vue/test-utils';
import AddView from '../index.vue';

describe('Test Add View', () => {
  // Nos devuelve un shallowmont del componente
  // es decir una copia de lo que se renderizaría en la web
  const build = () => {
    const wrapper = shallowMount(AddView);
    return {
      wrapper,
    };
  };

  //
  test('Debería ser igual a su Snapshot', () => {
    const { wrapper } = build();
    // El elemento debe coicidir con la copia montada
    expect(wrapper.element).toMatchSnapshot();
  });
  // Probamos método selectedPlant
  test('should selectedPlant', () => {
    const { wrapper } = build();
    wrapper.vm.selectedPlant();
  });
  // Probamos método clearItems
  test('should selectedPlant', () => {
    const { wrapper } = build();
    wrapper.vm.clearItems();
  });

  // Probamos método findPlant
  test('should selectedPlant', () => {
    const { wrapper } = build();
    wrapper.vm.findPlant('coco');
  });
  // Probamos método onSubmit
  test('should selectedPlant', () => {
    const { wrapper } = build();
    wrapper.vm.onSubmit('coco');
  });
});
