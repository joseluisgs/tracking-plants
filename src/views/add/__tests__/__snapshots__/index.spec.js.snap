// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Add View Deber√≠a ser igual a su Snapshot 1`] = `
<section
  class="mt-20 custom-container"
>
  <div
    class="flex justify-center"
  >
    <img
      alt="Cactus"
      class="w-48 h-64 rounded-xl"
      src=""
    />
  </div>
   
  <h1
    class="font-bold text-2xl text-center mt-2"
  >
    Nueva Planta
  </h1>
   
  <addplantform-stub />
</section>
`;
